FEXCore/Source/Interface/Core/CPUBackend.cpp

```diff
@@ -382,6 +382,20 @@ namespace CPU {
     if (static_cast<Context::ContextImpl*>(ThreadState->CTX)->Config.GlobalJITNaming()) {
       static_cast<Context::ContextImpl*>(ThreadState->CTX)->Symbols.RegisterJITSpace(Buffer.Ptr, Buffer.Size);
     }
+
+    // Protect the last page of the allocated buffer to trigger SIGSEGV on write access
+#ifndef _WIN32
+    const long pageSize = sysconf(_SC_PAGESIZE);
+    if (pageSize > 0 && Buffer.Size >= (size_t)pageSize) {
+      uintptr_t lastPageAddr = reinterpret_cast<uintptr_t>(Buffer.Ptr) + Buffer.Size - pageSize;
+      lastPageAddr = lastPageAddr & ~(pageSize - 1);
+
+      if (mprotect(reinterpret_cast<void*>(lastPageAddr), pageSize, PROT_NONE) != 0) {
```

Comment:
Use VirtualProtect here which is generic - and you'll probably want to adjust the CodeSize check to trigger a resize a page earlier